### PR TITLE
EVG-14601 Update statuses after tasks are aborted

### PR DESCRIPTION
**Pull request by bynn**
[EVG-14601](https://jira.mongodb.org/browse/EVG-14601)

### Description 
updates status on setActiveState now

### Testing 
https://spruce-staging.corp.mongodb.com/version/6137a8299dbe322c5de92d93/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC

### CODE REVIEW

File: model/task_lifecycle.go [hunk 1]
```diff
@@ -134,6 +134,7 @@ func SetActiveState(t *task.Task, caller string, active bool) error {
 		}
 	}
 
+	UpdateBuildAndVersionStatusForTask(t)
```
Comment (ablack12) [inline model/task_lifecycle.go:137]: We should return this so that we don't suppress the error

**Review by ablack12 — CHANGES_REQUESTED** [review]
Can you change this PR title to have an imperative verb describing what action you took (for example, "update build/version status with task activation")? The title you have here is the problem, not the solution, which we try to avoid.
https://docs.google.com/document/d/1LJHPjLb_bUZxv-l_9Y61kitzAg7u-13I5DjOhZ-deBA/edit#heading=h.fua24zxppw7l

File: model/task_lifecycle.go [hunk 2]
```diff
@@ -134,7 +134,10 @@ func SetActiveState(t *task.Task, caller string, active bool) error {
 		}
 	}
 
-	UpdateBuildAndVersionStatusForTask(t)
+	err := UpdateBuildAndVersionStatusForTask(t)
+	if err != nil {
```
Comment (ablack12) [inline model/task_lifecycle.go:138]: nit; combine in one line with 137 `if err = ... ; err != nil {`

**Review by ablack12 — APPROVED** [review]
LGTM with nit

File: model/task_lifecycle.go [hunk 3]
```diff
@@ -132,6 +132,10 @@ func SetActiveState(t *task.Task, caller string, active bool) error {
 		if err := UpdateDisplayTaskForTask(t); err != nil {
 			return errors.Wrap(err, "problem updating display task")
 		}
+	} else {
```
Comment (ablack12) [inline model/task_lifecycle.go:135]: Sorry for not catching this: I think we should do this regardless of IsPartOfDisplay; if the execution task status changed, the display task status will also have changed, so the build/version status will have changed.

**Review by ablack12 — CHANGES_REQUESTED** [review]

File: model/task_lifecycle.go [hunk 4]
```diff
@@ -128,6 +128,10 @@ func SetActiveState(t *task.Task, caller string, active bool) error {
 		return nil
 	}
 
+	if err := UpdateBuildAndVersionStatusForTask(t); err != nil {
```
Comment (ablack12) [inline model/task_lifecycle.go:131]: What was the logic for moving this above the IsPartOfDisplay check? I would think it wouldn't work right until the display task status is updated at 136.

Comment (bynn) [reply model/task_lifecycle.go]: yeah.

**Review by ablack12 — APPROVED** [review]

**Pull request merged**